user_tools/tests/spark_rapids_tools_e2e/resources/scripts/hdfs/setup_hdfs.sh

```diff
@@ -0,0 +1,177 @@
+    readonly E2E_TEST_DATA_NODE_DIR="${E2E_TEST_HDFS_DIR}/datanode"
+    rm -rf "${E2E_TEST_HDFS_DIR}" "${E2E_TEST_NAME_NODE_DIR}" "${E2E_TEST_DATA_NODE_DIR}"
+    mkdir -p "${E2E_TEST_HDFS_DIR}" "${E2E_TEST_NAME_NODE_DIR}" "${E2E_TEST_DATA_NODE_DIR}"
+    export E2E_TEST_NAME_NODE_DIR E2E_TEST_DATA_NODE_DIR
+}
+
+# Function to verify checksum
+verify_checksum() {
+    echo "Verifying checksum..."
+    if [ $# -ne 2 ]; then
+        err "verify_checksum requires two arguments: file and checksum_file."
+    fi
+    local file="$1"
+    local checksum_file="$2"
+    local expected_checksum=$(awk '{print $4}' "${checksum_file}")
+    local actual_checksum=$(shasum -a 512 "${file}" | awk '{print $1}')
+    [ "${expected_checksum}" != "${actual_checksum}" ] && return 1 || return 0
+}
+
+# Function to download and extract Hadoop
+download_and_extract_hadoop() {
+    echo "Downloading and extracting Hadoop..."
+    local hadoop_url="https://dlcdn.apache.org/hadoop/common/hadoop-${E2E_TEST_HADOOP_VERSION}/hadoop-${E2E_TEST_HADOOP_VERSION}.tar.gz"
+    local hadoop_tar_file="${E2E_TEST_TOOLS_TMP_DIR}/hadoop-${E2E_TEST_HADOOP_VERSION}.tar.gz"
+    local checksum_url="${hadoop_url}.sha512"
+    local checksum_file="${hadoop_tar_file}.sha512"
+
+    if [ ! -f "${hadoop_tar_file}" ]; then
+        wget -O "${hadoop_tar_file}" "${hadoop_url}" || err "Failed to download Hadoop tarball."
+    fi
+
+    # Verify checksum and re-download if needed
+    wget -O "${checksum_file}" "${checksum_url}" || err "Failed to download checksum file."
+    if ! verify_checksum "${hadoop_tar_file}" "${checksum_file}"; then
+        wget -O "${hadoop_tar_file}" "${hadoop_url}" || err "Failed to download Hadoop tarball."
+        if ! verify_checksum "${hadoop_tar_file}" "${checksum_file}"; then
+            err "Checksum verification failed after re-downloading. Exiting..."
+        fi
+    fi
+
+    tar -xzf "${hadoop_tar_file}" -C "${E2E_TEST_HDFS_DIR}" || err "Failed to extract Hadoop tarball."
+}
+
+# Configure Hadoop
+configure_hadoop() {
+    echo "Configuring Hadoop..."
+    readonly HADOOP_HOME="${E2E_TEST_HDFS_DIR}/hadoop-${E2E_TEST_HADOOP_VERSION}"
+    readonly HADOOP_CONF_DIR=$HADOOP_HOME/etc/hadoop
+    export HADOOP_HOME HADOOP_CONF_DIR
+    export PATH="${PATH}:${HADOOP_HOME}/bin:${HADOOP_HOME}/sbin"
+    envsubst < "${HDFS_SCRIPTS_DIR}/templates/${DEFAULT_CORE_SITE_XML}" > "${HADOOP_HOME}/etc/hadoop/core-site.xml"
+    envsubst < "${HDFS_SCRIPTS_DIR}/templates/${DEFAULT_HDFS_SITE_XML}" > "${HADOOP_HOME}/etc/hadoop/hdfs-site.xml"
+}
+
+# Format the Namenode
+format_namenode() {
+    echo "Formatting the Namenode..."
+    yes | hdfs namenode -format || err "Failed to format Namenode."
+}
+
+# Start HDFS services
+start_hdfs_services() {
+    echo "Starting HDFS services..."
+    hdfs --daemon start namenode
+    hdfs --daemon start datanode
+}
+
+# Verify that HDFS services are running
+verify_hdfs_services() {
+    echo "Verifying HDFS services..."
+    sleep 3
```

Comment:
3 could be short if system is very busy, might be nice to have a few retries in here or someway to specify.  I'm ok with waiting to change this as long as obvious from error messages.